Sources/Public/DotLottie/DotLottieFileHelpers.swift

```diff
@@ -268,4 +217,71 @@ extension DotLottieFile {
     }
   }
 
+  /// Loads an DotLottie from a specific filepath synchronously.
+  /// - Parameter filepath: The absolute filepath of the lottie to load. EG "/User/Me/starAnimation.lottie"
+  /// - Parameter dotLottieCache: A cache for holding loaded lotties. Defaults to `LRUDotLottieCache.sharedCache`. Optional.
+  public static func loadedFrom(
+    filepath: String,
+    dotLottieCache: DotLottieCacheProvider? = DotLottieCache.sharedCache)
+    -> Result<DotLottieFile, Error>
+  {
+    /// Check cache for lottie
+    if
+      let dotLottieCache = dotLottieCache,
+      let lottie = dotLottieCache.file(forKey: filepath)
+    {
+      return .success(lottie)
+    }
+
+    do {
+      /// Decode the lottie.
+      let url = URL(fileURLWithPath: filepath)
+      let data = try Data(contentsOf: url)
+      let lottie = try DotLottieFile(data: data, filename: url.deletingPathExtension().lastPathComponent)
```

Comment:
I don't think we should provide a synchronous method for loading DotLottie files. This initializer performs file system operations, like decompressing `data` to a temporary directly and reading various files. This would block the current thread, so we need to make sure we aren't making it too easy for folks to accidentally call this from the main thread.

One reason I think it's especially important to not provide synchronous loading methods for `DotLottieFile`s is because we _do_ provide synchronous loading methods for `LottieAnimation`s (which are less problematic to load on the main thread). If we provide synchronous methods for both I worry the difference would be too subtle.